tests/core-manager.js

```diff
@@ -457,6 +458,163 @@ test('unreplicate', async (t) => {
+    const dataWriter1 = cm1.getWriterCore('data')
+    const dataWriter2 = cm2.getWriterCore('data')
+
+    await dataWriter1.core.ready()
+    await dataWriter2.core.ready()
+
+    await dataWriter1.core.append(
+      Array(100)
+        .fill(null)
+        .map((_, i) => 'block' + i)
+    )
+
+    await dataWriter2.core.append(
+      Array(50)
+        .fill(null)
+        .map((_, i) => 'block' + i)
+    )
+
+    /// Replicate
+    const n1 = new NoiseSecretStream(true)
+    const n2 = new NoiseSecretStream(false)
+    n1.rawStream.pipe(n2.rawStream).pipe(n1.rawStream)
+    cm1[kCoreManagerReplicate](
+      // @ts-expect-error
+      n1
+    )
+    cm2[kCoreManagerReplicate](
+      // @ts-expect-error
+      n2
+    )
+
+    // This delay is needed in order for replication to finish properly
+    await new Promise((res) => setTimeout(res, 200))
+
+    /// Confirmation to ensure that replication worked
+    t.is(
+      cm1.getCores('data').length,
+      2,
+      'peer 1 has expected number of data cores after replication'
+    )
+
+    const peer1DataCoreStoragePath = getCoreStoragePath(
+      // @ts-expect-error
+      dataWriter1.core.discoveryKey.toString('hex')
+    )
+
+    const peer2DataCoreStoragePath = getCoreStoragePath(
+      // @ts-expect-error
+      dataWriter2.core.discoveryKey.toString('hex')
+    )
+
+    const dataCoreStorageNames = storageNames.filter(
+      (name) =>
+        name.startsWith(peer1DataCoreStoragePath) ||
+        name.startsWith(peer2DataCoreStoragePath)
+    )
+
+    // Hypercore uses 4 files per core (oplog, tree, bitfield, data)
+    t.is(
+      dataCoreStorageNames.length,
+      8,
+      'peer 1 has expected number of data core storage files after replication'
+    )
+
+    /// Delete data (not their own)
+    await t.test('with deleteOwn=false or omitted', async (st) => {
+      await cm1.deleteData('data')
+
+      const peer1DataStoragePreservedForPeer1 = (
+        await checkExistenceForFiles(
+          dataCoreStorageNames
+            .filter((storageName) =>
+              storageName.startsWith(peer1DataCoreStoragePath)
+            )
+            .map((storageName) => path.join(peer1TempPath, storageName))
+        )
+      ).every((exists) => exists === true)
+
+      const peer2DataStorageDeletedForPeer1 = (
+        await checkExistenceForFiles(
+          dataCoreStorageNames
+            .filter((storageName) =>
+              storageName.startsWith(peer2DataCoreStoragePath)
+            )
+            .map((storageName) => path.join(peer1TempPath, storageName))
+        )
+      ).every((exists) => exists === false)
+
+      st.ok(
+        peer1DataStoragePreservedForPeer1,
+        'peer 1 still has `data` storage for itself'
+      )
+
+      st.ok(
+        peer2DataStorageDeletedForPeer1,
+        'peer 1 no longer has `data` storage for peer 2'
+      )
+    })
+
+    /// Delete their own data
+    await t.test('with deleteOwn=true', async (st) => {
```

Comment:
thought it'd be easier to add this as part of the test but can split it out into its own test if that's preferred

Reply:
Fine for it to be here I think.